test/unittest/tests/test_tailq.cpp

```diff
@@ -0,0 +1,134 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <list>
+#include <catch2/catch.hpp>
+#include "wt_internal.h"
+
+template <class T> struct tailq_entry {
+    explicit tailq_entry<T>(T const &value) : _value(value){};
+    TAILQ_ENTRY(tailq_entry<T>) _queue;
+    T _value;
+};
+
+template <class T> class TestTailQWrapper {
+    public:
+    TestTailQWrapper();
+    ~TestTailQWrapper();
+    void pushBack(const T &value);
+    void pushBack(std::list<T> const &values);
```

Comment:
Same question if the `const` should be before `std::list`.

Reply:
`const <type>` and `<type> const` are equivalent. 
When references or pointers are involved, it's usually easier put `const` in the code in such a way that you can read it 'backwards' easily. So you can read that parameter as "reference to a const std::list". 
Which is also why I'd prefer our coding style allowed the `&` to be against the type, rather than against the identifier (ie `T& value` instead of `T &value`). 
For consistency, I should have declared `const T &value` as `T const &value` - I'll change that.